toolchain/sem_ir/value_stores.h

```diff
@@ -168,9 +168,22 @@ struct NameScope {
   // Names in the scope.
   llvm::DenseMap<NameId, InstId> names;
 
-  // Whether the scope corresponds to an import that failed. There may still be
-  // names from successful imports, or the current file.
-  bool has_load_error = false;
+  // Scopes extended by this scope.
+  //
+  // TODO: A `NameScopeId` is not sufficient to describe an extended scope in
+  // general. For example:
+  //
+  //   class A(T:! type) {
+  //     extend base: B(T*);
+  //   }
+  //
+  // needs to describe the `T*` argument.
+  llvm::SmallVector<NameScopeId, 1> extended_scopes;
```

Comment:
Is it worth noting this vector is examined in reverse order? (I assume more entries will be added down the road, although it's just one right now?)

Reply:
Thanks, nice catch. The order shouldn't affect whether name lookup succeeds, but it would affect which declarations we point to in ambiguity diagnostics (once we start producing ambiguity diagnostics...). Changed `LookupQualifiedName` to reverse these when they're pushed onto the queue. I think that'll mean we walk extended scopes in an unsurprising depth-first top-to-bottom preorder.